src/runtime_src/core/common/drv/xrt_cu.c

```diff
@@ -329,11 +350,10 @@ int xrt_cu_intr_thread(void *data)
 			continue;
 
 		if (xcu->num_sq || is_zero_credit(xcu)) {
-			//usleep_range(1, 3);
 			xrt_cu_check(xcu);
 			if (!xcu->done_cnt || !xcu->ready_cnt) {
 				xcu->sleep_cnt++;
-				if (down_interruptible(&xcu->sem_cu))
+				if (down_timeout(&xcu->sem_cu, CU_TIMER))
```

Comment:
I missed this change. Why do you need to change down_interruptible() to down_timeout()?

Reply:
I am not retrying this PR. Kindly post in XRT channel if you want this PR to be retried - Palash

Reply:
If a kernel runs for a long time or hang, ctrl-c cannot interrupt CU thread so this CU's c_cnt has no chance to increase.